### PR TITLE
Add support for Django 4.2 and Python 3.11.

### PR DESCRIPTION
**Pull request by codexterous**

### CODE REVIEW

**Comment by codexterous** [comment]
All errors appear to be related to "django.utils.deprecation.RemovedInDjango51Warning: django.contrib.auth.hashers.SHA1PasswordHasher is deprecated.", and are unrelated to this PR/not actually issues. For example:

```
  self = <django.contrib.auth.hashers.SHA1PasswordHasher object at 0x7f147ed810d0>
  args = (), kwargs = {}
  
      def __init__(self, *args, **kwargs):
  >       warnings.warn(
              "django.contrib.auth.hashers.SHA1PasswordHasher is deprecated.",
              RemovedInDjango51Warning,
              stacklevel=2,
          )
  E       django.utils.deprecation.RemovedInDjango51Warning: django.contrib.auth.hashers.SHA1PasswordHasher is deprecated.
```

**Comment by tiborhari** [comment]
Thanks for the change. Please also add `'3.11'` to the `python-version` list in `.github/workflows/tox.yml`, so that it'll be tested by GitHub actions.

**Comment by codexterous** [comment]
I went ahead and did that, but, I'm not normally a GitHub user and I'm not really familiar with GitHub actions. It seems to still be using python3.10 to run the test. Can you provide some guidance?

File: tox.ini [hunk 1]
```diff
@@ -40,6 +42,7 @@ python =
     3.8: py38
     3.9: py39
     3.10: py310, docs, package
+    3.11: py311, docs, package
```
Comment (tiborhari) [inline tox.ini]: Change this to `3.11: py311`, and it'll probably solve the build error.

**Pull request merged**